SwiftReflector/SwiftXmlReflection/ProtocolDeclaration.cs

```diff
@@ -48,6 +49,72 @@ public AssociatedTypeDeclaration AssociatedTypeNamed (string name)
 		{
 			return AssociatedTypes.FirstOrDefault (at => at.Name == name);
 		}
+
+		public bool HasDynamicSelf {
+			// you could cache this, but this type is not mutable, so that would be bad
+			get => SearchForDynamicSelf ();
+		}
+
+		bool SearchForDynamicSelf ()
+		{
+			foreach (var member in this.Members) {
+				if (member is FunctionDeclaration funcDecl) {
+					if (SearchForDynamicSelf (funcDecl))
+						return true;
+				} else if (member is PropertyDeclaration propDecl) {
+					if (SearchForDynamicSelf (propDecl)) {
+						return true;
+					}
+				} else {
+					throw new NotImplementedException ($"Unknown MemberDeclaration type {member.GetType ().Name}");
+				}
+			}
+			return false;
+		}
+
+		bool SearchForDynamicSelf (FunctionDeclaration funcDecl)
+		{
+			var types = funcDecl.ParameterLists.Last ().Select (p => p.TypeSpec).ToList ();
+			if (!TypeSpec.IsNullOrEmptyTuple (funcDecl.ReturnTypeSpec))
+				types.Add (funcDecl.ReturnTypeSpec);
+			return SearchForDynamicSelf (types);
+		}
+
+		bool SearchForDynamicSelf (PropertyDeclaration propDecl)
+		{
+			var types = new List<TypeSpec> ();
+			types.Add (propDecl.TypeSpec);
```

Comment:
that extraneous list (already in a loop, inside recursion) can be removed if you implement
`bool SearchForDynamicSelf (TypeSpec type)`
and then make
`bool SearchForDynamicSelf (List<TypeSpec> types)` use it